### DISCUSSION TITLE
Does update_topics always re extract words?

### DISCUSSION THREAD

**Discussion by zilch42** [MaartenGr/BERTopic#1882] [Q&A] [answered]
Quick question, because I can't seem to figure it out from the code... does `update_topics` always re-extract keywords and redo the ctfidf matrix regardless of the representation model e.g. if I've already got keywords I'm happy with and I just want to use the [DOCUMENTS] and [KEYWORDS] for an openAI representation model? 

Will this use the keywords already there or calculate them again?

```
# Topic model already fit...

tokenizer= tiktoken.encoding_for_model("gpt-3.5-turbo")
client = openai.OpenAI(api_key="sk-...")
representation_model = OpenAI(
    client,
    model="gpt-3.5-turbo", 
    delay_in_seconds=2, 
    chat=True,
    nr_docs=4,
    doc_length=100,
    tokenizer=tokenizer
)

topic_model.update_topics(docs, representation_model=representation_model)
```

**Comment by MaartenGr** [comment, marked as answer]
Yes, `.update_topics` will always re-calculate all representations. In your example, it will re-calculate the c-TF-IDF representation and then re-calculate the OpenAI representations. Having said that, if you use the same c-TF-IDF parameters as you did before, then it should provide the same keywords to the OpenAI model.
